### PR TITLE
Add CSP and X-Frame-Options in header response

### LINKED ISSUE #187: Disallow iframe
We should add a script to redirect iframe integration to full page.

### PR DESCRIPTION
**Pull request by gregory-latinier**
Close #187 

image

Now the response contains the Content Security Policies and X-Frame-Options to prevent some security issues

### CODE REVIEW

**Comment by bonustrack** [comment]
Working well for me, iframe is shown as blank.

**Comment by bonustrack** [comment]
Here is a link to deployment https://sc2-master-pr-201.herokuapp.com/ and here a jsfiddle to test iframe: https://jsfiddle.net/o6oohxbo/6/ @goldibex @sneak

File: app.js [hunk 1]
```diff
@@ -22,6 +23,25 @@ https.globalAgent.maxSockets = Infinity;
...
+app.use((req, res, next) => {
+  res.setHeader('X-Frame-Options', 'SAMEORIGIN');
+  next();
+});
+
+// Content security policies
+app.use(csp({
+  policies: {
+    'default-src': (process.env.CSP_DEFAULT || "'self'").split(','),
+    'script-src': (process.env.CSP_SCRIPT_SRC || "'self','unsafe-eval','unsafe-inline'").split(','),
```
Comment (goldibex) [inline app.js]: Please justify these settings for `script-src`

Comment (gregory-latinier) [reply app.js]: `'unsafe-eval'` and `'unsafe-line'` are set as default for the development environment
for production environment `CSP_SCRIPT_SRC` should only contain `'self'`

**Comment by bonustrack** [comment]
Is it fine to merge for you @goldibex?

**Comment by goldibex** [comment]
git 'er dun

**Comment by sneak** [comment]
In a perfect world, this would be using the `config` module like condenser does, and instead of having unsafe defaults hardcoded that someone must remember to override with sane values in production, the sane production defaults would be stored in `production.json` and the sane development defaults would be stored in `development.json`.

**Pull request merged**